### PR TITLE
[FIX] payment_worldline: various fixes

### PR DESCRIPTION
**Pull request by Demesmaeker**
Fixup of 65b273681db8277493ee399b3addc96fd284e100

Some transactions references are suffixed by a higher number than 0. For instance: _1, _2, ...

We now also manage that case.

----------------------------

Avoid sending False to the API

---------------------------

Send the customer's name to Worldline

### CODE REVIEW

File: addons/payment_worldline/models/payment_transaction.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def _process_notification_data(self, notification_data):
 
         # Update the provider reference.
         payment_data = notification_data['payment']
-        self.provider_reference = payment_data.get('id', '').removesuffix('_0')
+        self.provider_reference = payment_data.get('id', '').split('_', 1)[0]
```
Comment (AntoineVDV) [inline addons/payment_worldline/models/payment_transaction.py:278]: If you expect the `id` to contain more than one `_` (according to your `maxsplit=1` param), you should join all the strings resulting of a normal split except the last one. Here, you split on the first occurrence of `_`.

File: addons/payment_worldline/models/payment_transaction.py [hunk 2]
```diff
@@ -113,15 +113,15 @@ def _worldline_create_checkout_session(self):
                 },
                 'customer': {  # required to create a token and for some redirected payment methods
```
Comment (AntoineVDV) [inline addons/payment_worldline/models/payment_transaction.py:115]: `lang` is not required either.

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

File: addons/payment_worldline/models/payment_transaction.py (hunk 1 above)
Comment (Demesmaeker) [reply addons/payment_worldline/models/payment_transaction.py:278]: I'll change it as it makes sense, but we don't expect more than one AFAIK

File: addons/payment_worldline/models/payment_transaction.py (hunk 2 above)
Comment (Demesmaeker) [reply addons/payment_worldline/models/payment_transaction.py:115]: Would it be possible for us not to have one? :thinking:

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**